lib/dialects/oracledb.ts

```diff
@@ -125,57 +125,72 @@ export default class oracleDB implements SchemaInspector {
   columnInfo(table: string, column: string): Promise<Column>;
   async columnInfo<T>(table?: string, column?: string) {
     const query = this.knex
-      .with(
-        'uc',
-        this.knex.raw(`
-          SELECT /*+ materialize */ DISTINCT
-            "uc"."TABLE_NAME",
-            "ucc"."COLUMN_NAME",
-            "uc"."CONSTRAINT_NAME",
-            "uc"."CONSTRAINT_TYPE",
-            "uc"."R_CONSTRAINT_NAME"
-          FROM "USER_CONSTRAINTS" "uc"
-            INNER JOIN "USER_CONS_COLUMNS" "ucc" ON "uc"."CONSTRAINT_NAME" = "ucc"."CONSTRAINT_NAME"
-            AND "uc"."CONSTRAINT_TYPE" IN ('P', 'U', 'R')
-        `)
-      )
-      .select<RawColumn[]>(
-        'c.TABLE_NAME',
-        'c.COLUMN_NAME',
-        'c.DATA_DEFAULT',
-        'c.DATA_TYPE',
-        'c.DATA_LENGTH',
-        'c.DATA_PRECISION',
-        'c.DATA_SCALE',
-        'c.NULLABLE',
-        'c.IDENTITY_COLUMN',
-        'c.VIRTUAL_COLUMN',
-        'cm.COMMENTS as COLUMN_COMMENT',
-        'ct.CONSTRAINT_TYPE',
-        'fk.TABLE_NAME as REFERENCED_TABLE_NAME',
-        'fk.COLUMN_NAME as REFERENCED_COLUMN_NAME'
+      .select(
+        this.knex.raw(
```

Comment:
Why did we remove the `WITH` clause? This query was heavily optimized to be as performant as possible, since Oracle is incredibly slow by default.

I really don't want Oracle to take 30 seconds to load Directus the first time you log in again (that is not an exageration)!

Reply:
I though the `with` clause only made sense because it was joined against itself (so it probably benefited from the trick ) - I removed it because this is no longer the case in the implementation I propose. However there most definetly is room from improvement here if you want to help!

Reply:
I ran the tests on my machine  against `quillbuilduser/oracle-18-xe-micro-sq` which is configured in the compose file and in the test specs, and they completed without errors. I don't know why the tests fail in the CI...

Reply:
If the performance is there I have no objection to the new code. I can't remember exactly what the issue was, but I think without using `WITH` Oracle was just querying the `USER_CONSTRAINTS` excessively. But with `WITH` we could optimize that away by memorizing the results and accessing them in the main query.

If you want, you can remove the Oracle stuff from this PR and I'll add a fix tonight?

Reply:
OK, I just reverted the last commit and removed the failing index from Oracle' seed.

Reply:
Your refactored query clocks in at just under 4 seconds, compared to 1.3 seconds on the old query. Oracle is super shit at optimization. 😩

Reply:
Welp... hopefully it will help pave the way to a more efficient working solution 🙂